data/wp/wp-content/plugins/epfl-people/epfl-people.php

```diff
@@ -21,101 +24,61 @@ function epfl_people_log( $message ) {
     }
 }
 
-function epfl_people_url_exists( $url )
-{
-    $handle = curl_init( $url );
-    curl_setopt( $handle, CURLOPT_RETURNTRANSFER, TRUE );
-
-    $response = curl_exec( $handle );
-    $httpCode = curl_getinfo( $handle, CURLINFO_HTTP_CODE );
-
-    if ( $httpCode >= 200 && $httpCode <= 400 ) {
-        return true;
-    } else {
-        return false;
-    }
-    curl_close( $handle );
-}
-
 function epfl_people_process_shortcode( $attributes, $content = null )
 {
-    $attributes = shortcode_atts( array(
-        'url' => ''
+	
+   $attributes = shortcode_atts( array(
+        'unit' => ''
     ), $attributes );
-
-    // Sanitize parameter
-    $url = sanitize_text_field( $attributes['url'] );
-
-    // Check if the result is already in cache
-    $result = wp_cache_get( $url, 'epfl_people' );
-
-    if ( false === $result ){
-
-        // Make sure the content is actually coming from the people pages and does exist
-        if ( ( strcasecmp( parse_url( $url, PHP_URL_HOST ), 'people.epfl.ch' ) == 0 or strcasecmp( parse_url( $url, PHP_URL_HOST ), 'test-people.epfl.ch' ) == 0 ) && epfl_people_url_exists( $url ) ) {
-
-            // Get the content of the page
-            $response = wp_remote_get( $url );
-            $page = wp_remote_retrieve_body( $response );
-
-            // cache the result
-            wp_cache_set( $url, $page, 'epfl_people' );
-
-            // return the page
-            return '<div class="peopleListBox">'.
-                    $page.
-                    '</div>';
-        } else {
-            $error = new WP_Error( 'not found', 'The url passed is not part of people or is not found', $url );
-            epfl_people_log( $error );
-        }
-    } else {
-        // Use cache
-        return $result;
-    }
+   
+   // Sanitize parameter
+	$unit = sanitize_text_field( $attributes['unit'] );
+	
+	$url = "https://people.epfl.ch/cgi-bin/wsgetpeople?units=$unit&app=self&caller=104782";
```

Comment:
Pourquoi supposons-nous que Ion fait toutes les requêtes ? ☺ (Virer &caller= si pas de façon raisonnable de le deviner d'après le contexte)

Reply:
Les pulls requests commençant par [WIP] sont des work in progress, notre usage veut que l'on commente à la fin, une fois qu'elle est prête :)